### ISSUE TITLE
PlantUML import doesn't work in local network

### ISSUE THREAD

**Issue by romkavt** [plantuml/plantuml#1927]
**Describe the bug**
Starting PlantUML version 1.2024.5 and above, the import instruction stops working within corporate networks.
For example:
`!include https://bitbucket.yooteam.ru/pages/.../plantuml-library/c4-model.iuml`
produces the error "Cannot open URL".

The resource address is:

```
$ nslookup bitbucket.yooteam.ru
Non-authoritative answer:
bitbucket.yooteam.ru	canonical name = bitbucket.service.consul.yooteam.ru.
Name:	bitbucket.service.consul.yooteam.ru
Address: 10.225.90.10
```

The problem cause is https://github.com/plantuml/plantuml/blob/7ddfd61a328a5229f67f29bacb2f20e534ea641b/src/net/sourceforge/plantuml/security/URLCheck.java#L108
This code first introduced in version 1.2024.5.
This code resolves FQDN to IP address and verifies the address is not local network.

**To Reproduce**
Use PlantUML versions 1.2024.5 and above.
Use "import https:// ..." instruction referenced to any address in SiteLocal network space, i.e 10/8, 192.168/24 and so on.

**Expected behavior**
Allow import resources from local networks (10/8, 192.168/24 and so on)

**Comment by arnaudroques** [comment]
I think this is the same issue as https://github.com/plantuml/plantuml/issues/1820

You should either use `PLANTUML_ALLOWLIST_URL` or set your security profile to `UNSECURE` (see https://plantuml.com/security )

This should work with `V1.2024.7`

Hope this helps!

**Comment by arnaudroques** [comment]
Could you try with:

`export PLANTUML_SECURITY_PROFILE=UNSECURE
`

> Thanks, but this doesn't help in case of JetBrains IDEA + PlantUML plugin https://plugins.jetbrains.com/plugin/7017-plantuml-integration
> 
If it still does not work, you can open a ticket on JetBrains IDEA plugin team. We will explain them how to change this setting within their plugin. (You can show them this ticket so that we link both tickets)

> 
> I suggest to consider to remove local network address restriction or think how to integrate this into desktop tools. I think the local networks restriction is impractical for security.

Unfortunately, we had to set this restriction by default because PlantUML is used in many cases, including on websites exposed to the internet, and this could allow a hacker to use the server to test local addresses.

**Comment by romkavt** [comment]
> Unfortunately, we had to set this restriction by default because PlantUML is used in many cases, including on websites exposed to the internet, and this could allow a hacker to use the server to test local addresses.

I agree for direct IP-addresses restrictions, but, I think, FQDN request restrictions are impractical for security and makes some troubles for corporate users. I suggest to think again about that)

**Comment by BlackFIlms** [comment]
Hello, i have this trouble too.  
On latest version of docker container - **2025.2**  
Doesn't works security profile **UNSECURE**  

I try pass next variables to the container:  
- `PLANTUML_SECURITY_PROFILE=UNSECURE`  
-  `-DPLANTUML_SECURITY_PROFILE=UNSECURE`   

That doesn't resolving trouble. =/
